libazureinit/src/provision/user.rs

```diff
@@ -96,26 +96,103 @@ impl Provisioner {
     }
 }
 
+#[instrument(skip_all)]
+fn user_exists(username: &str) -> Result<bool, Error> {
+    let status = Command::new("getent")
+        .arg("passwd")
+        .arg(username)
+        .status()?;
+
+    Ok(status.success())
+}
+
 #[instrument(skip_all)]
 fn useradd(user: &User) -> Result<(), Error> {
+    if user_exists(&user.name)? {
+        tracing::info!(
+            "User '{}' already exists. Skipping user creation.",
+            user.name
+        );
+
+        let current_groups_output =
+            Command::new("id").arg("-nG").arg(&user.name).output()?;
+
+        let current_groups =
+            String::from_utf8_lossy(&current_groups_output.stdout)
+                .split_whitespace()
+                .map(String::from) // Convert &str to String
+                .collect::<HashSet<_>>();
+
+        let new_groups =
+            user.groups.iter().cloned().collect::<HashSet<String>>();
+
+        let groups_to_add = new_groups.difference(&current_groups);
+
+        if !groups_to_add.clone().collect::<Vec<_>>().is_empty() {
+            let group_list = groups_to_add
+                .map(|group| group.as_str())
+                .collect::<Vec<_>>()
+                .join(",");
+
+            tracing::info!(
+                "User '{}' is missing some groups. Adding them to the following groups: {}",
+                user.name,
+                group_list
+            );
+
+            let usermod_command =
+                format!("usermod -aG {} {}", group_list, user.name);
+
+            tracing::debug!("Running command: {}", usermod_command);
```

Comment:
This is fine for this PR but long term speaking, we should create a wrapper for Command that will log all commands (we need to pay attention to sensitive data and redact them appropriately)

Reply:
I think we already have an issue open for improved logging messages (if not, I'll open one), but I will move that up the priority list in terms of what is next.